### PR TITLE
[SYCL] Fix sycl::vec unary ops

### PR DESCRIPTION
**Pull request by cperkinsintel**
The recent sycl::vec changes (https://github.com/intel/llvm/pull/9492)  broke they unary operations. This PR fixes them and adds some testing to avoid that in the future.

### CODE REVIEW

**Comment by cperkinsintel** [comment]
The unary operator `!` is not supported for `std::byte` ( https://godbolt.org/z/7eWTKT9xP ).  So it seems to me that it shouldn't be supported for `sycl::vec<std::byte>` either. I'm a little confused as to how it ever worked.  
In any case, the `Basic/vector_byte.cpp` test is failing with this PR. The simplest solution would be to remove the unary test at line 293 of that file, but I'm not sure how acceptable that would be.

File: sycl/test/basic_tests/types.cpp [hunk 1]
```diff
@@ -101,6 +101,67 @@ template <> inline void checkSizeForFloatingPoint<s::half, sizeof(int16_t)>() {
...
+
+  std::cout << vec2string(v) << std::endl;
+
+  T d = +v;
+  std::cout << vec2string(d) << std::endl;
+
+  T e = -v;
+  std::cout << vec2string(e) << std::endl;
+
+  // ~ only supported by integral types.
```
Comment (cperkinsintel) [inline sycl/test/basic_tests/types.cpp:127]: I should update this. Not supported by floating point types is the correct formulation. std::byte is neither integral nor floating, but the unary operator~() is expected to work with it.

**Comment by bader** [comment]
std::byte supports explicit conversion to an integer type (and back) - https://en.cppreference.com/w/cpp/types/byte#Non-member_functions.

https://godbolt.org/z/3z65E4Th4

**Comment by cperkinsintel** [comment]
ping to reviewers. This PR has been updated and tests are passing.

**Review by dm-vodopyanov — APPROVED** [review]

**Pull request merged**